src/EnergyPlus/UnitarySystem.cc

```diff
@@ -1088,28 +1088,52 @@ namespace UnitarySystems {
 
         if (this->m_MySetPointCheckFlag) {
             if (!state.dataGlobal->SysSizingCalc && state.dataHVACGlobal->DoSetPointTest) {
-
+                bool setPointMissing = false;
                 if (this->m_CoolCoilExists) {
                     int ControlNode = this->m_SystemCoolControlNodeNum;
                     if (ControlNode > 0) {
                         this->checkNodeSetPoint(state, AirLoopNum, ControlNode, CoolingCoil, OAUCoilOutTemp);
+                    } else if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
+                        ShowSevereError(state, format("{}: Missing cooling set point in {} = {}", routineName, this->UnitType, this->Name));
+                        ShowContinueError(state,
+                                          format("...Setpoint is required at system air outlet node = {} or cooling coil air outlet node = {}",
+                                                 state.dataLoopNodes->NodeID(this->AirOutNode),
+                                                 state.dataLoopNodes->NodeID(this->CoolCoilOutletNodeNum)));
+                        setPointMissing = true;
                     }
                 }
 
                 if (this->m_HeatCoilExists) {
                     int ControlNode = this->m_SystemHeatControlNodeNum;
                     if (ControlNode > 0) {
                         this->checkNodeSetPoint(state, AirLoopNum, ControlNode, HeatingCoil, OAUCoilOutTemp);
+                    } else if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
+                        ShowSevereError(state, format("{}: Missing heating set point in {} = {}", routineName, this->UnitType, this->Name));
+                        ShowContinueError(state,
+                                          format("...Setpoint is required at system air outlet node = {} or heating coil air outlet node = {}",
+                                                 state.dataLoopNodes->NodeID(this->AirOutNode),
+                                                 state.dataLoopNodes->NodeID(this->HeatCoilOutletNodeNum)));
+                        setPointMissing = true;
```

Comment:
Alert user when set point does not exist (ControlNode <= 0) for set point controlled coil.

Reply:
This works ok, but it seems this could be incorporated into `checkNodeSetpoint` , it could check first for a zero control node and then throw the error.